src/lib/Behat/BusinessContext/AdministrationContext.php

```diff
@@ -176,18 +177,37 @@ public function iGoToListItem(string $itemName, string $itemType, string $itemCo
      */
     public function iStartEditingItem(string $itemType, string $itemName, ?string $containerName = null): void
     {
-        PageObjectFactory::createPage($this->utilityContext, $this->itemCreateMapping[$itemType], $containerName)
-            ->adminList->table->clickEditButton($itemName);
+        $pageItemName = $containerName;
+        $pageType = $this->itemCreateMapping[$itemType];
+        if($containerName === 'details') {
+            $pageItemName = $itemName;
+            $pageType = $itemType;
+        }
+        PageObjectFactory::createPage($this->utilityContext, $pageType, $pageItemName)
+            ->startEditingItem($itemName);
     }
 
     /**
-     * @When I delete :itemType :itemName
-     * @When I delete :itemType :itemName from :itemContainer
+     * @When I delete :itemType
+     * @When I delete :itemType from :containerName
      */
-    public function iDeleteItem(string $itemType, string $itemName, ?string $itemContainer = null): void
```

Comment:
Same here: I'd split it into 3 steps (I delete, I delete from container, I delete from details page). I don't think we need to have such complicated steps.